### ISSUE TITLE
Error when using old saved dashboards or question with custom columns, not working

### ISSUE THREAD

**Issue by juangc-wavyssa** [metabase/metabase#36893]
**Describe the bug**
When I open a report or question that had custom columns configured before, it no longer works.

**Logs**
Please include javascript console and server logs around the time this bug occurred. For information about how to get these, consult our [bug troubleshooting guide](https://metabase.com/docs/latest/troubleshooting-guide/bugs.html)

[23f75950-2312-480a-87c8-e4de22f5ae73] 2023-12-18T11:42:16+01:00 DEBUG metabase.server.middleware.log POST /api/dataset 202 [ASYNC: completed] 648,7 ms (18 llamadas a la BBDD) Conexiones BBDD: 1/15 Hilos Jetty: 2/50 (5 esperando, 0 en cola) (106 hilos totales activos) Consultas en ejecución: 0 (0 en cola); postgres DB 2 connections: 0/1 (0 threads blocked)
[23f75950-2312-480a-87c8-e4de22f5ae73] 2023-12-18T11:42:40+01:00 ERROR metabase.query-processor.middleware.catch-exceptions Error procesando consulta: Entrada inválida: {:query {:aggregation [nil [nil [nil ["valid instance of one of these MBQL clauses: :expression, :field"]]]]}}
{:database_id 2,
 :started_at #t "2023-12-18T10:42:40.109557Z[GMT]",
 :action_id nil,
 :json_query
 {:database 2,
  :type "query",
  :query
  {:aggregation
   [["sum" ["field" 2629 {:base-type "type/Decimal"}]]
    ["aggregation-options" ["avg"] {:name "LOS media", :display-name "LOS media"}]],
   :breakout
   [["field" 989 {:base-type "type/Text", :join-alias "Hotel"}]
    ["field" 556 {:base-type "type/Date", :temporal-unit "month"}]],
   :joins
   [{:alias "Hotel",
     :fields "all",
     :condition
     ["=" ["field" 562 {:base-type "type/Integer"}] ["field" 990 {:base-type "type/Integer", :join-alias "Hotel"}]],
     :source-table 73}],
   :source-table 44,
   :filter
   ["and"
    ["=" ["field" 550 {:base-type "type/PostgresEnum"}] "CANCELLED"]
    ["time-interval" ["field" 556 {:base-type "type/Date"}] 3 "month"]
    ["time-interval" ["field" 581 {:base-type "type/Date"}] -7 "day"]]},
  :parameters [],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> query_processor.middleware.fetch_source_query$fn__67673$fn__67678.invoke(fetch_source_query.clj:278)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__67681.invoke(fetch_source_query.clj:295)"
  "query_processor.middleware.resolve_database_and_driver$resolve_driver_and_database_local_values$fn__73760$fn__73764.invoke(resolve_database_and_driver.clj:77)"
  "driver$do_with_driver.invokeStatic(driver.clj:94)"
  "driver$do_with_driver.invoke(driver.clj:89)"
  "query_processor.middleware.resolve_database_and_driver$resolve_driver_and_database_local_values$fn__73760.invoke(resolve_database_and_driver.clj:76)"
  "query_processor.middleware.store$initialize_store$fn__68098$fn__68099.invoke(store.clj:14)"
  "query_processor.store$do_with_metadata_provider.invokeStatic(store.clj:169)"
  "query_processor.store$do_with_metadata_provider.invoke(store.clj:150)"
  "query_processor.store$do_with_metadata_provider.invokeStatic(store.clj:158)"
  "query_processor.store$do_with_metadata_provider.invoke(store.clj:150)"
  "query_processor.middleware.store$initialize_store$fn__68098.invoke(store.clj:13)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database$fn__73757.invoke(resolve_database_and_driver.clj:60)"
  "query_processor.middleware.normalize_query$normalize$fn__74067.invoke(normalize_query.clj:38)"
  "query_processor.middleware.enterprise$fn__73687$handle_audit_app_internal_queries__73688$fn__73690.invoke(enterprise.clj:96)"
  "query_processor.middleware.enterprise$handle_audit_app_internal_queries_middleware$fn__73698.invoke(enterprise.clj:103)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__70671.invoke(constraints.clj:81)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__73998.invoke(process_userland_query.clj:156)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__74596.invoke(catch_exceptions.clj:171)"
  "query_processor.reducible$async_qp$qp_STAR___62528$thunk__62530.invoke(reducible.clj:126)"
  "query_processor.reducible$async_qp$qp_STAR___62528.invoke(reducible.clj:132)"
  "query_processor.reducible$sync_qp$qp_STAR___62540.doInvoke(reducible.clj:153)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:401)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:397)"
  "query_processor$process_query_and_save_execution_BANG_.invokeStatic(query_processor.clj:415)"
  "query_processor$process_query_and_save_execution_BANG_.invoke(query_processor.clj:405)"
  "query_processor$process_query_and_save_with_max_results_constraints_BANG_.invokeStatic(query_processor.clj:430)"
  "query_processor$process_query_and_save_with_max_results_constraints_BANG_.invoke(query_processor.clj:420)"
  "api.dataset$run_query_async$fn__97301.invoke(dataset.clj:79)"
  "query_processor.streaming$streaming_response_STAR_$fn__51831$fn__51833.invoke(streaming.clj:168)"
  "query_processor.streaming$streaming_response_STAR_$fn__51831.invoke(streaming.clj:167)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:69)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:67)"
  "async.streaming_response$do_f_async$task__43034.invoke(streaming_response.clj:88)"],
 :card_id nil,
 :context :ad-hoc,
 :error
 "Entrada inválida: {:query {:aggregation [nil [nil [nil [\"valid instance of one of these MBQL clauses: :expression, :field\"]]]]}}",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data
 {:type :metabase.util.malli.fn/invalid-input,
  :error
  {:schema [:ref :metabase.mbql.schema/Query],
   :value
   {:database 2,
    :type :query,
    :query
    {:aggregation
     [[:sum [:field 2629 {:base-type :type/Decimal}]]
      [:aggregation-options [:avg nil] {:name "LOS media", :display-name "LOS media"}]],
     :breakout
     [[:field 989 {:base-type :type/Text, :join-alias "Hotel"}]
      [:field 556 {:base-type :type/Date, :temporal-unit :month}]],
     :joins
     [{:alias "Hotel",
       :fields :all,
       :condition
       [:= [:field 562 {:base-type :type/Integer}] [:field 990 {:base-type :type/Integer, :join-alias "Hotel"}]],
       :source-table 73}],
     :source-table 44,
     :filter
     [:and
      [:= [:field 550 {:base-type :type/PostgresEnum}] "CANCELLED"]
      [:time-interval [:field 556 {:base-type :type/Date}] 3 :month]
      [:time-interval [:field 581 {:base-type :type/Date}] -7 :day]]},
    :middleware {:js-int-to-string? true, :add-default-userland-constraints? true},
    :info {:executed-by 11, :context :ad-hoc, :query-hash #object["[B" 0x9ae6213 "[B@9ae6213"]},
    :constraints {:max-results 10000, :max-results-bare-rows 2000}},
   :errors
   ({:path
     [0
      0
      0
      :query
      0
      0
      :aggregation
      0
      0
      :aggregation-options
      0
      1
      "aggregation"
      0
      :else
      :avg
      0
      1
      "field-or-expression"
      :else
      0],
     :in [:query :aggregation 1 1 1],
     :schema
     [:multi {:dispatch #object[metabase.mbql.schema.helpers$clause_tag 0x12713530 "metabase.mbql.schema.helpers$clause_tag@12713530"], :error/message "valid instance of one of these MBQL clauses: :expression, :field"} [:expression [:ref :metabase.mbql.schema/expression]] [:field [:ref :metabase.mbql.schema/field]]],
     :value nil,
     :type :malli.core/invalid-dispatch-value})},
  :humanized
  {:query {:aggregation [nil [nil [nil ["valid instance of one of these MBQL clauses: :expression, :field"]]]]}},
  :value
  {:database 2,
   :type :query,
   :query
   {:aggregation
    [[:sum [:field 2629 {:base-type :type/Decimal}]]
     [:aggregation-options [:avg nil] {:name "LOS media", :display-name "LOS media"}]],
    :breakout
    [[:field 989 {:base-type :type/Text, :join-alias "Hotel"}]
     [:field 556 {:base-type :type/Date, :temporal-unit :month}]],
    :joins
    [{:alias "Hotel",
      :fields :all,
      :condition
      [:= [:field 562 {:base-type :type/Integer}] [:field 990 {:base-type :type/Integer, :join-alias "Hotel"}]],
      :source-table 73}],
    :source-table 44,
    :filter
    [:and
     [:= [:field 550 {:base-type :type/PostgresEnum}] "CANCELLED"]
     [:time-interval [:field 556 {:base-type :type/Date}] 3 :month]
     [:time-interval [:field 581 {:base-type :type/Date}] -7 :day]]},
   :middleware {:js-int-to-string? true, :add-default-userland-constraints? true},
   :info {:executed-by 11, :context :ad-hoc, :query-hash #object["[B" 0x9ae6213 "[B@9ae6213"]},
   :constraints {:max-results 10000, :max-results-bare-rows 2000}},
  :fn-name metabase.query-processor.middleware.fetch-source-query/resolve-card-id-source-tables*},
 :data {:rows [], :cols []}}

[23f75950-2312-480a-87c8-e4de22f5ae73] 2023-12-18T11:42:40+01:00 DEBUG metabase.server.middleware.log POST /api/dataset 202 [ASYNC: completed] 223,6 ms (15 llamadas a la BBDD) Conexiones BBDD: 1/15 Hilos Jetty: 2/50 (5 esperando, 0 en cola) (105 hilos totales activos) Consultas en ejecución: 0 (0 en cola)
[23f75950-2312-480a-87c8-e4de22f5ae73] 2023-12-18T11:43:36+01:00 WARN metabase.lib.convert Clean: Removing bad clause in [:expressions 0] due to error {:expressions [["invalid MBQL clause: [:- {:lib/uuid \"d09dcce8-725c-4b6b-8049-bc7e40d3608f\", :lib/expression-name \"LOS\"} [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid \"b0baeb58-25a7-43e9-9cae-9734d04ba4d3\"} 576] [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid \"9f6073a6-416b-4f15-a776-9db1739b6727\"} 572]]" "Invalid :- clause: Temporal arithmetic expression must contain exactly one non-interval value"]]}:
{:expressions
 [[:-
   {:lib/uuid "d09dcce8-725c-4b6b-8049-bc7e40d3608f", :lib/expression-name "LOS"}
   [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid "b0baeb58-25a7-43e9-9cae-9734d04ba4d3"} 576]
   [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid "9f6073a6-416b-4f15-a776-9db1739b6727"} 572]]]}

[23f75950-2312-480a-87c8-e4de22f5ae73] 2023-12-18T11:43:36+01:00 WARN metabase.lib.convert Clean: Removing bad clause in [:expressions] due to error :malli.core/limits:
{:expressions []}

[23f75950-2312-480a-87c8-e4de22f5ae73] 2023-12-18T11:43:37+01:00 WARN metabase.lib.convert Clean: Removing bad clause in [:expressions 0] due to error {:expressions [["invalid MBQL clause: [:- {:lib/uuid \"c2b7bcd1-120b-40d1-a0b9-daf63998157e\", :lib/expression-name \"LOS\"} [:field {:base-type :type/Date, :temporal-unit :default, :metabase.query-processor.util.add-alias-info/source-table 44, :metabase.query-processor.util.add-alias-info/source-alias \"departure_date\", :metabase.query-processor.util.add-alias-info/desired-alias \"departure_date\", :metabase.query-processor.util.add-alias-info/position 5, :lib/uuid \"6eb95011-c511-4183-bb28-6c7f9402aec6\"} 576] [:field {:base-type :type/Date, :temporal-unit :default, :metabase.query-processor.util.add-alias-info/source-table 44, :metabase.query-processor.util.add-alias-info/source-alias \"arrival_date\", :metabase.query-processor.util.add-alias-info/desired-alias \"arrival_date\", :metabase.query-processor.util.add-alias-info/position 4, :lib/uuid \"dcfffecc-e98f-45b8-a3ff-6c3ec297e379\"} 572]]" "Invalid :- clause: Temporal arithmetic expression must contain exactly one non-interval value"]]}:
{:expressions
 [[:-
   {:lib/uuid "c2b7bcd1-120b-40d1-a0b9-daf63998157e", :lib/expression-name "LOS"}
   [:field
    {:base-type :type/Date,
     :temporal-unit :default,
     :metabase.query-processor.util.add-alias-info/source-table 44,
     :metabase.query-processor.util.add-alias-info/source-alias "departure_date",
     :metabase.query-processor.util.add-alias-info/desired-alias "departure_date",
     :metabase.query-processor.util.add-alias-info/position 5,
     :lib/uuid "6eb95011-c511-4183-bb28-6c7f9402aec6"}
    576]
   [:field
    {:base-type :type/Date,
     :temporal-unit :default,
     :metabase.query-processor.util.add-alias-info/source-table 44,
     :metabase.query-processor.util.add-alias-info/source-alias "arrival_date",
     :metabase.query-processor.util.add-alias-info/desired-alias "arrival_date",
     :metabase.query-processor.util.add-alias-info/position 4,
     :lib/uuid "dcfffecc-e98f-45b8-a3ff-6c3ec297e379"}
    572]]]}

[23f75950-2312-480a-87c8-e4de22f5ae73] 2023-12-18T11:43:37+01:00 WARN metabase.lib.convert Clean: Removing bad clause in [:expressions] due to error :malli.core/limits:
{:expressions []}

[23f75950-2312-480a-87c8-e4de22f5ae73] 2023-12-18T11:43:37+01:00 WARN metabase.lib.convert Clean: Removing bad clause in [:expressions 0] due to error {:expressions [["invalid MBQL clause: [:- {:lib/uuid \"a4cb4867-3944-4647-988d-fb8bcdeb6bd8\", :lib/expression-name \"LOS\"} [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid \"0caa60c0-cb15-4d39-849a-cb421a9d56c5\"} 576] [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid \"e21c2440-a1ad-4811-a6e1-f13223c8410e\"} 572]]" "Invalid :- clause: Temporal arithmetic expression must contain exactly one non-interval value"]]}:
{:expressions
 [[:-
   {:lib/uuid "a4cb4867-3944-4647-988d-fb8bcdeb6bd8", :lib/expression-name "LOS"}
   [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid "0caa60c0-cb15-4d39-849a-cb421a9d56c5"} 576]
   [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid "e21c2440-a1ad-4811-a6e1-f13223c8410e"} 572]]]}

[23f75950-2312-480a-87c8-e4de22f5ae73] 2023-12-18T11:43:37+01:00 WARN metabase.lib.convert Clean: Removing bad clause in [:expressions] due to error :malli.core/limits:
{:expressions []}

[23f75950-2312-480a-87c8-e4de22f5ae73] 2023-12-18T11:43:37+01:00 ERROR metabase.query-processor.middleware.catch-exceptions Error procesando consulta: No hay ninguna expresión con el nombre "LOS"
{:database_id 2,
 :started_at #t "2023-12-18T10:43:36.845847Z[GMT]",
 :action_id nil,
 :json_query
 {:database 2,
  :type "query",
  :query
  {:aggregation
   [["sum" ["field" 2629 {:base-type "type/Decimal"}]]
    ["aggregation-options"
     ["avg" ["expression" "LOS" {:base-type "type/Date"}]]
     {:name "LOS media", :display-name "LOS media"}]],
   :breakout
   [["field" 989 {:base-type "type/Text", :join-alias "Hotel"}]
    ["field" 556 {:base-type "type/Date", :temporal-unit "month"}]],
   :joins
   [{:alias "Hotel",
     :fields "all",
     :condition
     ["=" ["field" 562 {:base-type "type/Integer"}] ["field" 990 {:base-type "type/Integer", :join-alias "Hotel"}]],
     :source-table 73}],
   :source-table 44,
   :expressions {:LOS ["-" ["field" 576 {:base-type "type/Date"}] ["field" 572 {:base-type "type/Date"}]]},
   :filter
   ["and"
    ["=" ["field" 550 {:base-type "type/PostgresEnum"}] "CANCELLED"]
    ["time-interval" ["field" 556 {:base-type "type/Date"}] 3 "month"]
    ["time-interval" ["field" 581 {:base-type "type/Date"}] -7 "day"]]},
  :parameters [],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native
 {:query
  "SELECT \"source\".\"Hotel__name\" AS \"Hotel__name\", DATE_TRUNC('month', \"source\".\"stay_date\") AS \"stay_date\", SUM(\"source\".\"room_net_revenue\") AS \"sum\", AVG(\"source\".\"LOS\") AS \"LOS media\" FROM (SELECT \"satocan\".\"reservation_lines\".\"hotel_id\" AS \"hotel_id\", \"satocan\".\"reservation_lines\".\"stay_date\" AS \"stay_date\", \"satocan\".\"reservation_lines\".\"status\" AS \"status\", \"satocan\".\"reservation_lines\".\"cancellation_date\" AS \"cancellation_date\", \"satocan\".\"reservation_lines\".\"arrival_date\" AS \"arrival_date\", \"satocan\".\"reservation_lines\".\"departure_date\" AS \"departure_date\", \"satocan\".\"reservation_lines\".\"room_net_revenue\" AS \"room_net_revenue\", \"satocan\".\"reservation_lines\".\"departure_date\" - \"satocan\".\"reservation_lines\".\"arrival_date\" AS \"LOS\", \"Hotel\".\"id\" AS \"Hotel__id\", \"Hotel\".\"name\" AS \"Hotel__name\", \"Hotel\".\"acronym\" AS \"Hotel__acronym\", \"Hotel\".\"enabled\" AS \"Hotel__enabled\", \"Hotel\".\"creation_date\" AS \"Hotel__creation_date\", \"Hotel\".\"modification_date\" AS \"Hotel__modification_date\", \"Hotel\".\"address\" AS \"Hotel__address\", \"Hotel\".\"order\" AS \"Hotel__order\", \"Hotel\".\"zone\" AS \"Hotel__zone\" FROM \"satocan\".\"reservation_lines\" LEFT JOIN \"satocan\".\"hotel\" AS \"Hotel\" ON \"satocan\".\"reservation_lines\".\"hotel_id\" = \"Hotel\".\"id\" WHERE (\"satocan\".\"reservation_lines\".\"status\" = CAST(? AS \"reservation_status\")) AND (\"satocan\".\"reservation_lines\".\"stay_date\" >= DATE_TRUNC('month', (NOW() + INTERVAL '1 month'))) AND (\"satocan\".\"reservation_lines\".\"stay_date\" < DATE_TRUNC('month', (NOW() + INTERVAL '4 month'))) AND (\"satocan\".\"reservation_lines\".\"cancellation_date\" >= CAST((NOW() + INTERVAL '-7 day') AS date)) AND (\"satocan\".\"reservation_lines\".\"cancellation_date\" < CAST(NOW() AS date))) AS \"source\" GROUP BY \"source\".\"Hotel__name\", DATE_TRUNC('month', \"source\".\"stay_date\") ORDER BY \"source\".\"Hotel__name\" ASC, DATE_TRUNC('month', \"source\".\"stay_date\") ASC",
  :params ("CANCELLED")},
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> lib.expression$fn__54541$_AMPERSAND_f__54542.invoke(expression.cljc:46)"
  "lib.expression$fn__54541$fn__54545.invoke(expression.cljc:34)"
  "lib.expression$fn__54549.invokeStatic(expression.cljc:53)"
  "lib.expression$fn__54549.invoke(expression.cljc:51)"
  "lib.metadata.calculation$fn__54253$_AMPERSAND_f__54254.invoke(calculation.cljc:184)"
  "lib.metadata.calculation$fn__54253$fn__54261.invoke(calculation.cljc:161)"
  "lib.expression$fn__54555.invokeStatic(expression.cljc:63)"
  "lib.expression$fn__54555.invoke(expression.cljc:56)"
  "lib.metadata.calculation$fn__54293$_AMPERSAND_f__54294.invoke(calculation.cljc:248)"
  "lib.metadata.calculation$fn__54293$fn__54296.invoke(calculation.cljc:233)"
  "lib.aggregation$fn__56921.invokeStatic(aggregation.cljc:204)"
  "lib.aggregation$fn__56921.invoke(aggregation.cljc:198)"
  "lib.metadata.calculation$fn__54293$_AMPERSAND_f__54294.invoke(calculation.cljc:248)"
  "lib.metadata.calculation$fn__54293$fn__54296.invoke(calculation.cljc:233)"
  "query_processor.middleware.annotate$fn__64869$_AMPERSAND_f__64870.invoke(annotate.clj:352)"
  "query_processor.middleware.annotate$fn__64869$fn__64872.invoke(annotate.clj:333)"
  "query_processor.middleware.annotate$fn__64904$_AMPERSAND_f__64906$iter__64921__64925$fn__64926$fn__64927.invoke(annotate.clj:401)"
  "query_processor.middleware.annotate$fn__64904$_AMPERSAND_f__64906$iter__64921__64925$fn__64926.invoke(annotate.clj:400)"
  "query_processor.middleware.annotate$maybe_merge_source_metadata.invokeStatic(annotate.clj:436)"
  "query_processor.middleware.annotate$maybe_merge_source_metadata.invoke(annotate.clj:431)"
  "query_processor.middleware.annotate$mbql_cols.invokeStatic(annotate.clj:479)"
  "query_processor.middleware.annotate$mbql_cols.invoke(annotate.clj:466)"
  "query_processor.middleware.annotate$fn__65022.invokeStatic(annotate.clj:500)"
  "query_processor.middleware.annotate$fn__65022.invoke(annotate.clj:496)"
  "query_processor.middleware.annotate$fn__65041$_AMPERSAND_f__65043.invoke(annotate.clj:571)"
  "query_processor.middleware.annotate$fn__65041$fn__65046.invoke(annotate.clj:565)"
  "query_processor.middleware.annotate$add_column_info$add_column_info_rff_STAR___65080.invoke(annotate.clj:621)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__72812$fn__72813.invoke(mbql_to_native.clj:27)"
  "query_processor.context.default$default_reducef.invokeStatic(default.clj:31)"
  "query_processor.context.default$default_reducef.invoke(default.clj:25)"
  "query_processor.context$reducef.invokeStatic(context.clj:70)"
  "query_processor.context$reducef.invoke(context.clj:63)"
  "query_processor.context.default$default_runf$respond_STAR___49661.invoke(default.clj:45)"
  "driver.sql_jdbc.execute$execute_reducible_query$fn__81066.invoke(execute.clj:706)"
  "driver.sql_jdbc.execute$fn__80861$fn__80862.invoke(execute.clj:388)"
  "driver.sql_jdbc.execute$fn__80826$_AMPERSAND_f__80827.invoke(execute.clj:334)"
  "driver.sql_jdbc.execute$fn__80826$fn__80830.invoke(execute.clj:317)"
  "driver.sql_jdbc.execute$fn__80861.invokeStatic(execute.clj:382)"
  "driver.sql_jdbc.execute$fn__80861.invoke(execute.clj:380)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:689)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:678)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:686)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:678)"
  "driver.sql_jdbc$fn__109931.invokeStatic(sql_jdbc.clj:82)"
  "driver.sql_jdbc$fn__109931.invoke(sql_jdbc.clj:80)"
  "query_processor.context$executef.invokeStatic(context.clj:60)"
  "query_processor.context$executef.invoke(context.clj:49)"
  "query_processor.context.default$default_runf.invokeStatic(default.clj:44)"
  "query_processor.context.default$default_runf.invoke(default.clj:42)"
  "query_processor.context$runf.invokeStatic(context.clj:46)"
  "query_processor.context$runf.invoke(context.clj:40)"
  "query_processor.reducible$identity_qp.invokeStatic(reducible.clj:39)"
  "query_processor.reducible$identity_qp.invoke(reducible.clj:36)"
  "query_processor.middleware.cache$maybe_return_cached_results$maybe_return_cached_results_STAR___73849.invoke(cache.clj:229)"
  "query_processor.middleware.permissions$check_query_permissions$fn__67554.invoke(permissions.clj:140)"
  "query_processor.middleware.enterprise$check_download_permissions_middleware$fn__73660.invoke(enterprise.clj:51)"
  "query_processor.middleware.enterprise$maybe_apply_column_level_perms_check_middleware$fn__73670.invoke(enterprise.clj:64)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__72812.invoke(mbql_to_native.clj:24)"
  "query_processor$fn__75052$combined_post_process__75057$combined_post_process_STAR___75058.invoke(query_processor.clj:261)"
  "query_processor$fn__75052$combined_pre_process__75053$combined_pre_process_STAR___75054.invoke(query_processor.clj:258)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__67681.invoke(fetch_source_query.clj:303)"
  "query_processor.middleware.resolve_database_and_driver$resolve_driver_and_database_local_values$fn__73760$fn__73764.invoke(resolve_database_and_driver.clj:77)"
  "driver$do_with_driver.invokeStatic(driver.clj:94)"
  "driver$do_with_driver.invoke(driver.clj:89)"
  "query_processor.middleware.resolve_database_and_driver$resolve_driver_and_database_local_values$fn__73760.invoke(resolve_database_and_driver.clj:76)"
  "query_processor.middleware.store$initialize_store$fn__68098$fn__68099.invoke(store.clj:14)"
  "query_processor.store$do_with_metadata_provider.invokeStatic(store.clj:169)"
  "query_processor.store$do_with_metadata_provider.invoke(store.clj:150)"
  "query_processor.store$do_with_metadata_provider.invokeStatic(store.clj:158)"
  "query_processor.store$do_with_metadata_provider.invoke(store.clj:150)"
  "query_processor.middleware.store$initialize_store$fn__68098.invoke(store.clj:13)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database$fn__73757.invoke(resolve_database_and_driver.clj:60)"
  "query_processor.middleware.normalize_query$normalize$fn__74067.invoke(normalize_query.clj:38)"
  "query_processor.middleware.enterprise$fn__73687$handle_audit_app_internal_queries__73688$fn__73690.invoke(enterprise.clj:96)"
  "query_processor.middleware.enterprise$handle_audit_app_internal_queries_middleware$fn__73698.invoke(enterprise.clj:103)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__70671.invoke(constraints.clj:81)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__73998.invoke(process_userland_query.clj:156)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__74596.invoke(catch_exceptions.clj:171)"
  "query_processor.reducible$async_qp$qp_STAR___62528$thunk__62530.invoke(reducible.clj:126)"
  "query_processor.reducible$async_qp$qp_STAR___62528.invoke(reducible.clj:132)"
  "query_processor.reducible$sync_qp$qp_STAR___62540.doInvoke(reducible.clj:153)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:401)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:397)"
  "query_processor$process_query_and_save_execution_BANG_.invokeStatic(query_processor.clj:415)"
  "query_processor$process_query_and_save_execution_BANG_.invoke(query_processor.clj:405)"
  "query_processor$process_query_and_save_with_max_results_constraints_BANG_.invokeStatic(query_processor.clj:430)"
  "query_processor$process_query_and_save_with_max_results_constraints_BANG_.invoke(query_processor.clj:420)"
  "api.dataset$run_query_async$fn__97301.invoke(dataset.clj:79)"
  "query_processor.streaming$streaming_response_STAR_$fn__51831$fn__51833.invoke(streaming.clj:168)"
  "query_processor.streaming$streaming_response_STAR_$fn__51831.invoke(streaming.clj:167)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:69)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:67)"
  "async.streaming_response$do_f_async$task__43034.invoke(streaming_response.clj:88)"],
 :card_id nil,
 :context :ad-hoc,
 :error "No hay ninguna expresión con el nombre \"LOS\"",
 :row_count 0,
 :running_time 0,
 :preprocessed
 {:database 2,
  :type :query,
  :query
  {:aggregation
   [[:aggregation-options [:sum [:field 2629 {:base-type :type/Decimal}]] {:name "sum"}]
    [:aggregation-options [:avg [:expression "LOS"]] {:name "LOS media", :display-name "LOS media"}]],
   :breakout
   [[:field 989 {:base-type :type/Text, :join-alias "Hotel"}]
    [:field 556 {:base-type :type/Date, :temporal-unit :month}]],
   :source-table 44,
   :expressions
   {"LOS"
    [:-
     [:field 576 {:base-type :type/Date, :temporal-unit :default}]
     [:field 572 {:base-type :type/Date, :temporal-unit :default}]]},
   :filter
   [:and
    [:=
     [:field 550 {:base-type :type/PostgresEnum}]
     [:value
      "CANCELLED"
      {:base_type :type/PostgresEnum,
       :effective_type :type/PostgresEnum,
       :coercion_strategy nil,
       :semantic_type :type/Category,
       :database_type "reservation_status",
       :name "status"}]]
    [:>= [:field 556 {:base-type :type/Date, :temporal-unit :default}] [:relative-datetime 1 :month]]
    [:< [:field 556 {:base-type :type/Date, :temporal-unit :default}] [:relative-datetime 4 :month]]
    [:>= [:field 581 {:base-type :type/Date, :temporal-unit :default}] [:relative-datetime -7 :day]]
    [:< [:field 581 {:base-type :type/Date, :temporal-unit :default}] [:relative-datetime 0 :day]]],
   :order-by
   [[:asc [:field 989 {:base-type :type/Text, :join-alias "Hotel"}]]
    [:asc [:field 556 {:base-type :type/Date, :temporal-unit :month}]]],
   :joins
   [{:alias "Hotel",
     :strategy :left-join,
     :fields
     [[:field 990 {:join-alias "Hotel"}]
      [:field 989 {:join-alias "Hotel"}]
      [:field 991 {:join-alias "Hotel"}]
      [:field 992 {:join-alias "Hotel"}]
      [:field 988 {:temporal-unit :default, :join-alias "Hotel"}]
      [:field 995 {:temporal-unit :default, :join-alias "Hotel"}]
      [:field 994 {:join-alias "Hotel"}]
      [:field 993 {:join-alias "Hotel"}]
      [:field 2599 {:join-alias "Hotel"}]],
     :condition
     [:= [:field 562 {:base-type :type/Integer}] [:field 990 {:base-type :type/Integer, :join-alias "Hotel"}]],
     :source-table 73}]},
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true},
  :info {:executed-by 11, :context :ad-hoc}},
 :ex-data
 {:expression-name "LOS",
  :query
  {:lib/type :mbql/query,
   :stages
   [{:lib/type :mbql.stage/mbql,
     :aggregation
     [[:sum
       {:lib/uuid "4572979c-ced0-4894-b0ea-d40c53aff270", :name "sum"}
       [:field {:base-type :type/Decimal, :lib/uuid "5f7e5663-4bb7-41cb-8b16-86fca758d3a8"} 2629]]
      [:avg {:lib/uuid "06485b74-e074-4e09-81de-95f4410fa11e", :name "LOS media", :display-name "LOS media"}]],
     :breakout
     [[:field {:base-type :type/Text, :join-alias "Hotel", :lib/uuid "d9b1c7ba-fc55-40a6-8e61-fa322d264d7a"} 989]
      [:field {:base-type :type/Date, :temporal-unit :month, :lib/uuid "24e68ef9-51d9-44fc-b8e1-eeb3f3fd5325"} 556]],
     :source-table 44,
     :order-by
     [[:asc
       {:lib/uuid "611a30eb-51f3-4031-bf14-2170552f3b30"}
       [:field {:base-type :type/Text, :join-alias "Hotel", :lib/uuid "599186da-eb43-4d97-a829-9101fa406cc0"} 989]]
      [:asc
       {:lib/uuid "cb4c82cc-f9da-45d5-9207-7bbdad5fb02f"}
       [:field {:base-type :type/Date, :temporal-unit :month, :lib/uuid "1f9f15d7-cbd8-454f-9914-389134ce019b"} 556]]],
     :joins
     [{:alias "Hotel",
       :strategy :left-join,
       :fields
       [[:field
         {:base-type :type/Integer,
          :effective-type :type/Integer,
          :join-alias "Hotel",
          :lib/uuid "91db2e6e-b287-4f4f-80ff-a2c671dcb927"}
         990]
        [:field
         {:base-type :type/Text,
          :effective-type :type/Text,
          :join-alias "Hotel",
          :lib/uuid "eb466245-6f05-40b6-a0ad-75fad0c314cd"}
         989]
        [:field
         {:base-type :type/Text,
          :effective-type :type/Text,
          :join-alias "Hotel",
          :lib/uuid "5b659cc1-bcf5-468e-98d4-681b54c11978"}
         991]
        [:field
         {:base-type :type/Boolean,
          :effective-type :type/Boolean,
          :join-alias "Hotel",
          :lib/uuid "9d7481e9-2005-4947-aa04-b6030c37b24c"}
         992]
        [:field
         {:base-type :type/DateTimeWithLocalTZ,
          :effective-type :type/DateTimeWithLocalTZ,
          :temporal-unit :default,
          :join-alias "Hotel",
          :lib/uuid "13cbc8bf-4fa7-4013-865b-fba3096d6d59"}
         988]
        [:field
         {:base-type :type/DateTimeWithLocalTZ,
          :effective-type :type/DateTimeWithLocalTZ,
          :temporal-unit :default,
          :join-alias "Hotel",
          :lib/uuid "6b217f52-79d5-4603-99ad-9b1728f6c7ba"}
         995]
        [:field
         {:base-type :type/Text,
          :effective-type :type/Text,
          :join-alias "Hotel",
          :lib/uuid "729fdf51-044a-414b-90ef-901dbc81594b"}
         994]
        [:field
         {:base-type :type/Integer,
          :effective-type :type/Integer,
          :join-alias "Hotel",
          :lib/uuid "150619a5-d20c-40a9-bf0f-bf16bdd759f9"}
         993]
        [:field
         {:base-type :type/Text,
          :effective-type :type/Text,
          :join-alias "Hotel",
          :lib/uuid "4f4e5325-a718-4d52-87cd-8c518f0ee083"}
         2599]],
       :conditions
       [[:=
         {:lib/uuid "999df5e6-73bd-4bba-98fe-1db3e16de98d"}
         [:field {:base-type :type/Integer, :lib/uuid "4989a002-753f-42c1-9c5a-8b08fc80e4e7"} 562]
         [:field
          {:base-type :type/Integer, :join-alias "Hotel", :lib/uuid "97b22112-e59d-425b-9c8a-613fb22e9d41"}
          990]]],
       :lib/type :mbql/join,
       :stages [{:lib/type :mbql.stage/mbql, :source-table 73}],
       :lib/options {:lib/uuid "20ffc315-c62b-46bd-8ffc-849d23d31645"}}],
     :filters
     [[:=
       {:lib/uuid "e6e7b5e5-6a7a-42f7-bc99-4e6f6d63e854"}
       [:field {:base-type :type/PostgresEnum, :lib/uuid "30918bbf-0be8-41b7-b060-229b64623e10"} 550]
       [:value
        {:effective_type :type/PostgresEnum,
         :name "status",
         :base-type :type/PostgresEnum,
         :semantic-type :type/Category,
         :database-type "reservation_status",
         :effective-type :type/PostgresEnum,
         :lib/uuid "309f4fac-2feb-47ae-8273-5b043ecda769"}
        "CANCELLED"]]
      [:>=
       {:lib/uuid "77ddb4cd-f70b-4aaf-961a-d8c11d96afae"}
       [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid "998a3b60-1825-46e5-8580-a563cfd103cd"} 556]
       [:relative-datetime {:lib/uuid "deed932b-f956-44a4-95e5-b779ba905ab8"} 1 :month]]
      [:<
       {:lib/uuid "04c8d139-84f3-4838-a37a-cc8bb4c7d65d"}
       [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid "1e2eed89-6414-4f0b-be2e-cf16ddf6e969"} 556]
       [:relative-datetime {:lib/uuid "d90a1deb-70d9-4c16-a080-5337463aee10"} 4 :month]]
      [:>=
       {:lib/uuid "161f128f-bad6-48de-9031-5d06ae2d7e81"}
       [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid "464d1d59-e49f-4601-9afe-971ca216ebaf"} 581]
       [:relative-datetime {:lib/uuid "5f692783-8692-4f19-bd01-732b5c7e7e89"} -7 :day]]
      [:<
       {:lib/uuid "be64116b-8f65-44cc-bba2-93933649e914"}
       [:field {:base-type :type/Date, :temporal-unit :default, :lib/uuid "ed2ed4bd-6427-401b-b645-5680af0509ac"} 581]
       [:relative-datetime {:lib/uuid "4873da84-c23d-4cd0-b789-0cf84e46f09a"} 0 :day]]]}],
   :database 2,
   :lib/metadata
   (metabase.lib.metadata.cached-provider/cached-metadata-provider (metabase.lib.metadata.jvm/->UncachedApplicationDatabaseMetadataProvider 2))},
  :stage-number -1},
 :data {:rows [], :cols []}}

**To Reproduce**
1. Go to any old question with custom column and use custom column

**Expected behavior**
I expect that custom column works properly and shows info.

**Screenshots**

**Severity**
This is blocking our usage of Metabase regarding some dashboards that we use regularly that have custom columns.

<img width="1727" alt="Captura de pantalla 2023-12-18 a las 11 58 29" src="https://github.com/metabase/metabase/assets/112938617/fbd4904c-6459-4f42-94fc-191c620c6443">
<img width="1727" alt="Captura de pantalla 2023-12-18 a las 11 58 36" src="https://github.com/metabase/metabase/assets/112938617/d890f53e-4e7f-4e93-83b9-ac45d8ce7e96">
<img width="1727" alt="Captura de pantalla 2023-12-18 a las 11 58 44" src="https://github.com/metabase/metabase/assets/112938617/2041c250-4b1f-485c-b2fd-3d83e87a3f93">
<img width="1727" alt="Captura de pantalla 2023-12-18 a las 11 59 12" src="https://github.com/metabase/metabase/assets/112938617/3637228f-afc8-4eb9-b775-b51fac40206d">
<img width="1727" alt="Captura de pantalla 2023-12-18 a las 11 59 18" src="https://github.com/metabase/metabase/assets/112938617/73fef82c-8633-41b9-a3c1-711c2605cabc">

**Additional context**
When using custom field this error shows:

Salida no válida: {:stages [{:expressions [["invalid MBQL clause: [:- {:lib/uuid \"6f9d0415-bb04-4f56-bf6f-efff4159bea4\", :lib/expression-name \"LOS\"} [:field {:base-type :type/Date, :effective-type :type/Date, :temporal-unit :default, :lib/uuid \"abb4a42b-9588-4668-9172-d2e2f14fbcbe\"} 576] [:field {:base-type :type/Date, :effective-type :type/Date, :temporal-unit :default, :lib/uuid \"7e1b2dbe-0a13-4a18-ad1e-68da594c5440\"} 572]]" "Invalid :- clause: Temporal arithmetic expression must contain exactly one non-interval value" "invalid MBQL clause: [:- {:lib/uuid \"6f9d0415-bb04-4f56-bf6f-efff4159bea4\", :lib/expression-name \"LOS\"} [:field {:base-type :type/Date, :effective-type :type/Date, :temporal-unit :default, :lib/uuid \"abb4a42b-9588-4668-9172-d2e2f14fbcbe\"} 576] [:field {:base-type :type/Date, :effective-type :type/Date, :temporal-unit :default, :lib/uuid \"7e1b2dbe-0a13-4a18-ad1e-68da594c5440\"} 572]]" "Invalid :- clause: Temporal arithmetic expression must contain exactly one non-interval value"]], :source-card ["missing required key"]}]}

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "es-ES",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.21+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.21",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.21+9",
    "os.name": "Linux",
    "os.version": "4.14.252-195.483.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.14"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.6.0"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-12-14",
      "tag": "v0.48.0",
      "hash": "f985e19"
    },
    "settings": {
      "report-timezone": "Atlantic/Canary"
    }
  }
}
```

**Comment by bshepherdson** [comment]
This should not be throwing exceptions anymore after #36942. That fix is in 48.1, so I'm closing this issue.
Please reopen if this issue is not resolved on 48.1.

**Comment by juangc-wavyssa** [comment]
We see that the error persists when there are dates in the custom column, for example:
![image-20231226-154440](https://github.com/metabase/metabase/assets/112938617/f37bf5f8-d5b6-4170-8a80-f7a8bbcf2a09)
![image-20231226-154452](https://github.com/metabase/metabase/assets/112938617/c8b80de8-b53b-49a6-a6ee-12ee4e632dc1)

When we click on "visualizar" (visualize):

![image-20231226-154533](https://github.com/metabase/metabase/assets/112938617/abadc086-097e-487c-836b-c4cdce2e018b)

...and I go back to the report settings and see:
![image-20231226-154606](https://github.com/metabase/metabase/assets/112938617/2f4043c0-96de-40a0-89f0-4ecfd90e70c7)

In a report, when accessing it through a URL, it loads when you click, but if you go to the editor without making any changes and then click on "visualizar" (visualize), it gives an error.


Please, reopen the issue.

**Comment by darksciencebase** [comment]
@juangc-wavyssa do i understand correctly that you've updated to 48.1 and the issue is still there?

**Comment by juangc-wavyssa** [comment]
Yes!

**Comment by bshepherdson** [comment]
I think this is a different problem not really related to the original issue.

In both `master` and 48.1 expressions like `= [End Date] - [Start Date]` don't work correctly. I'm working on a patch to fix that properly (as a time `Interval`) but as a workaround you can use the built in date function `datetimeDiff`.

Here's a similar query. `Order History - Order -> Status Date` is a clunky auto-generated name; it should be `Delivery Date`. For your query the expression would be `= datetimeDiff([Departure Date], [Arrival Date], "day")`.

![2024-01-04-103420_1469x1185_scrot](https://github.com/metabase/metabase/assets/157812/ee4292a9-a20a-436f-a35b-304dfdb41f0d)

![2024-01-04-103433_1044x767_scrot](https://github.com/metabase/metabase/assets/157812/4eea8fdf-9b44-4c4e-b2d3-6631327572ec)

**Comment by bshepherdson** [comment]
I've filed #37263 to track this separate issue; I'm closing this one.
